src/util.c

```diff
@@ -297,6 +308,22 @@ unsigned long decodeFile(const char *filename, unsigned char **image, enum filet
     return ret;
 }
 
+unsigned long decodeFileFromBuffer(unsigned char *buf, long bufSize, unsigned char **image, enum filetype type, int *width, int *height, int pixelFormat) {
```

Comment:
ditto that `decodeFile` should call `decodeFileFromBuffer`

Reply:
Done